[WORKFLOW] use dev as release draft branch trigger

0xdavinchee opened the pull request.

0xdavinchee committed:
use dev as release draft branch

hellwolf commented:
Are we missing anything obvious?

hellwolf commented:
Incremental improvementalist inside me is screaming for a shared utility shell script that does the following while are at it, if you hold on, I could take a stab at it too:

```
          # sdk-redux
          PUBLISHED_SDK_REDUX_VERSION=$(npm show @superfluid-finance/sdk-redux@latest version)
          NEW_SDK_REDUX_VERSION=$(jq -r .version packages/sdk-redux/package.json)
          if [ "$PUBLISHED_SDK_REDUX_VERSION" != "$NEW_SDK_REDUX_VERSION" ];then
            echo "PUBLISHED_SDK_REDUX_VERSION=1" >> "$GITHUB_ENV"
            echo "sdk-redux changed: $PUBLISHED_SDK_REDUX_VERSION -> $NEW_SDK_REDUX_VERSION"
          else
            echo sdk-redux unchanged: "$PUBLISHED_SDK_REDUX_VERSION"
          fi

          echo "NEW_SDK_REDUX_VERSION=$NEW_SDK_REDUX_VERSION" >> "$GITHUB_OUTPUT"
```

0xdavinchee committed:
check package version test for sdk-core

0xdavinchee commented:
> Are we missing anything obvious?

Nothing other than making the utility shell script shared so we reduce some duplication.

> Incremental improvementalist inside me is screaming for a shared utility shell script that does the following while are at it, if you hold on, I could take a stab at it too:
> 
> ```
>           # sdk-redux
>           PUBLISHED_SDK_REDUX_VERSION=$(npm show @superfluid-finance/sdk-redux@latest version)
>           NEW_SDK_REDUX_VERSION=$(jq -r .version packages/sdk-redux/package.json)
>           if [ "$PUBLISHED_SDK_REDUX_VERSION" != "$NEW_SDK_REDUX_VERSION" ];then
>             echo "PUBLISHED_SDK_REDUX_VERSION=1" >> "$GITHUB_ENV"
>             echo "sdk-redux changed: $PUBLISHED_SDK_REDUX_VERSION -> $NEW_SDK_REDUX_VERSION"
>           else
>             echo sdk-redux unchanged: "$PUBLISHED_SDK_REDUX_VERSION"
>           fi
> 
>           echo "NEW_SDK_REDUX_VERSION=$NEW_SDK_REDUX_VERSION" >> "$GITHUB_OUTPUT"
> ```

pushed a file and giving this a try in a workflow run right now 

cc @hellwolf

hellwolf committed:
refactor check-package-version more

hellwolf commented:
Consider to use the new script for auto release of trivial package such as metadata, see https://github.com/superfluid-finance/protocol-monorepo/issues/1366#issuecomment-1628931113

0xdavinchee committed:
one workflow for package releases

- move all workflow files in
- next step:
  - try reusable workflow for the common steps

0xdavinchee committed:
fix actionlint error

0xdavinchee committed:
cleanup

hellwolf commented on .github/workflows/cd.sdk-redux-stable.create-release-drafts.yml:
```diff
@@ -6,7 +6,7 @@ on:
   workflow_dispatch:
```
not removed?

0xdavinchee committed:
wip

- remove sdk-redux create release draft
- use template release notes file

0xdavinchee committed:
no dashes

0xdavinchee committed:
Merge branch 'dev' into dev-on-push-release-draft-cd

0xdavinchee committed:
add metadata

- add metadata release + publish in create-release-drafts
- add LICENSE to metadata
- update monorepo README.md

hellwolf commented on .github/workflows/cd.packages-stable.create-release-drafts.yml:
```diff
@@ -0,0 +1,196 @@
+          release_name: metadata@v${{ env.METADATA_NEW_VERSION }}
+          body: |
+            Please refer to the CHANGELOG.md file in metadata package for more details on the changeset.
+          draft: true
```
do we still want to keep it draft stage? Since it's going to be published to npm nonethless.

0xdavinchee replied:
true, will set this to false

hellwolf replied:
that's default I believe

0xdavinchee committed:
set draft: false

hellwolf reviewed (approved):
ggg
